comp/metadata/softwareinventory/impl/inventorysoftware.go

```diff
@@ -134,48 +143,112 @@ func newWithClient(reqs Requires, client sysProbeClient) (Provides, error) {
 		log:            reqs.Log,
 		sysProbeClient: client,
 		hostname:       hname,
+		eventPlatform:  reqs.EventPlatform,
 	}
 
-	// Note that there is a second way to disable this feature, through InventoryPayload.Enabled.
-	// 'enable_metadata_collection' and 'inventories_enabled' both need to be set to true.
 	if !reqs.Config.GetBool("software_inventory.enabled") {
 		return Provides{
 			Comp: is,
 		}, nil
 	}
 
+	localSource := rand.NewSource(time.Now().UnixNano())
+	localRand := rand.New(localSource)
+
+	is.jitter = time.Duration(localRand.Intn(max(reqs.Config.GetInt("software_inventory.jitter"), 60))) * time.Second
+	is.interval = time.Duration(max(reqs.Config.GetInt("software_inventory.interval"), 10)) * time.Minute
+
 	is.log.Infof("Starting the inventory software component")
-	is.InventoryPayload = util.CreateInventoryPayload(reqs.Config, reqs.Log, reqs.Serializer, is.getPayload, flareFileName)
+
+	// Perform initial collection
+	installedSoftware, err := is.sysProbeClient.GetCheck(sysconfig.SoftwareInventoryModule)
```

Comment:
Just gaining context on this PR out of curiosity - 

it looks like we currently update the inventory cache whenever we call `getPayload`, whereas with these changes, we do it every tick. 

What is the difference in the frequency we update the cache now?